src/app/pages/admin/components/add-movie/add-movie.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { ToastrService } from 'ngx-toastr';
+import { IMovie, MoviesService } from '../../../../core/services/movies.service';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-add-movie',
+  templateUrl: './add-movie.component.html',
+  styleUrls: ['./add-movie.component.scss'],
+})
+export class AddMovieComponent implements OnInit, OnDestroy {
+  movieForm: FormGroup;
+
+  subscription: Subscription;
+
+  mouseoverLogin: boolean;
+
+  private preview: FormControl;
+
+  private title: FormControl;
+
+  private description: FormControl;
+
+  private genre: FormControl;
+
+  constructor(private readonly fb: FormBuilder, private readonly toastr: ToastrService,
+    private readonly movieService: MoviesService, private readonly router: Router) {
+  }
+
+  ngOnInit(): void {
+    this.preview = new FormControl('', [Validators.required]);
+    this.title = new FormControl('', [Validators.required]);
+    this.description = new FormControl('', [Validators.required]);
+    this.genre = new FormControl('', [Validators.required]);
```

Comment:
Look like you are not reusing these vars, so would be better to delete them for now and add in the feature if necessary